GTG/gtk/editor/editor.py

```diff
@@ -184,46 +194,113 @@ def __init__(self,
+
+    #     # Ctrl-I marks task as dismissed
+    #     dismiss_editor = self.builder.get_object('dismiss_editor')
+    #     key, mod = Gtk.accelerator_parse('<Control>i')
+    #     dismiss_editor.add_accelerator('clicked', agr, key, mod,
+    #                                    Gtk.AccelFlags.VISIBLE)
+
+    #     # Ctrl-Q quits GTG
+    #     key, modifier = Gtk.accelerator_parse('<Control>q')
+    #     agr.connect(key, modifier, Gtk.AccelFlags.VISIBLE, self.quit)
+
+    '''
+    TODO(jakubbrindza): Add the functionality to the existing calendar widgets.
+    This will require ammending and re-factoring the entire calendar.py.
+    '''
+
+    def show_popover_start(self, widget, popover):
+        self.popover = self.builder.get_object("date_popover")
+        self.popover.set_relative_to(self.builder.get_object
+                                     ("startdate_entry"))
+        self.popover.set_modal(False)
+        # popover.hide(self.builder.get_object("fuzzy_btns"))
+        self.popover.show_all()
+
+    def show_popover_due(self, widget, popover):
+        self.popover = self.builder.get_object("date_popover")
+        self.popover.set_relative_to(self.builder.get_object("duedate_entry"))
+        self.popover.set_modal(False)
+        # popover.hide(self.builder.get_object("fuzzy_btns"))
+        self.popover.show_all()
+
+    def show_popover_closed(self, widget, popover):
+        self.popover = self.builder.get_object("closed_popover")
+        self.popover.set_relative_to(self.builder.get_object
+                                     ("closeddate_entry"))
+        self.popover.set_modal(False)
+        # popover.hide(self.builder.get_object("fuzzy_btns"))
+        self.popover.show_all()
+
+    def open_tags_popover(self, widget):
+        if self.store is None:
+            self.store = Gtk.ListStore(bool, str)
+            self.store.clear()
+
+            self.treeview = self.builder.get_object("tags_tree")
+
+            renderer_toggle = Gtk.CellRendererToggle()
+            renderer_toggle.connect("toggled", self.on_cell_toggled)
+            column_toggle = Gtk.TreeViewColumn("Toggle", renderer_toggle,
+                                               active=0)
+            self.treeview.append_column(column_toggle)
+
+            renderer_text = Gtk.CellRendererText()
+            column_text = Gtk.TreeViewColumn("Text", renderer_text, text=1)
+            self.treeview.append_column(column_text)
+
+            tags = self.req.get_tag_tree().get_all_nodes()
+
+            used_tags = self.task.get_tags()
+
+            for tagname in tags:
+                tag = self.req.get_tag(tagname)
+                if tag_filter(tag):
+                    is_used = tag in used_tags
+                    self.store.append([is_used, tagname])
+
+            self.treeview.set_model(self.store)
+            self.treeview.show_all()
+
+    def on_cell_toggled(self, widget, path):
```

Comment:
can you name this method better? Something like `on_tag_toggled`? There could be potentially many cells here.